mobile/android/android-device-file-system.ts

```diff
@@ -54,8 +54,13 @@ export class AndroidDeviceFileSystem implements Mobile.IDeviceFileSystem {
 		await Promise.all(
 			_(localToDevicePaths)
 				.filter(localToDevicePathData => this.$fs.getFsStats(localToDevicePathData.getLocalPath()).isFile())
-				.map(async localToDevicePathData =>
-					await this.adb.executeCommand(["push", localToDevicePathData.getLocalPath(), localToDevicePathData.getDevicePath()])
+				.map(async localToDevicePathData => {
+					const devicePath = localToDevicePathData.getDevicePath();
+					let sections = devicePath.split("/");
+					let parentDir = sections.slice(0, sections.length - 1);
```

Comment:
instead of splitting and joining, I believe you can use `path.dirname(devicePath)` - have you tried it?